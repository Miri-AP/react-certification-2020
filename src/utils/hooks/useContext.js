import { createContext } from 'react';
import { color } from '../globalStyle';

export const initalState = {
  header: {
    background: color.oceanBlue,
    input: color.lightCoral,
  },
  card: {
    background: color.lightCoral,
    textColor: color.white,
  },
  background: `${color.white}`,
};

export const StyleContext = createContext(initalState);

export const VideoListContext = createContext([]);

// style Reducer
export const reducer = (state, action) => {
  switch (action) {
    case 'light':
      return initalState;
    case 'dark':
      return {
        header: {
          background: color.lightCoral,
          input: color.oceanBlue,
        },
        card: {
          background: color.seaTan,
          textColor: color.oceanBlue,
        },
        background: color.oceanBlue,
      };
    default:
      throw new Error();
  }
};
