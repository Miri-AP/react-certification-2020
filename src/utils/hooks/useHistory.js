import { useState } from 'react';

export const useHistory = () => {
  const [history, setHistory] = useState('');
  function setHistoryVal(keyword) {
    setHistory(keyword);
  }

  return { history, setHistoryVal };
};

// export { useHistory };
