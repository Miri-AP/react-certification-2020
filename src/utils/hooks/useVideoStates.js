import { useState, useEffect } from 'react';
import youtubeVideoList from '../mock/youtube-videos-mock.json';

export function useVideList() {
  console.log('called useVideList sa...');
  const [videoList, setVideoList] = useState([]);
  const [error, setError] = useState(null);

  useEffect(() => {
    const getList = async () => {
      let listItems = [];

      try {
        listItems = await youtubeVideoList.items;
      } catch (e) {
        console.info('Error reading json file: youtubeVideoList');
        setError(e);
      }
      setVideoList(listItems);
    };
    getList();
  }, []);

  function updateVideoList(videos) {
    setVideoList(videos);
  }

  function setErrorMsg(errorMsg) {
    setError(errorMsg);
  }

  return { videoList, error, setErrorMsg, updateVideoList };
}

export function useVideoInfo() {
  const [video, setVideo] = useState({});

  function updateVideoInfo(data) {
    setVideo(data);
  }

  return { video, updateVideoInfo };
}
