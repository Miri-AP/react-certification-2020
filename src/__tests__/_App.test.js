import React from 'react';
import { screen, cleanup, render } from '@testing-library/react';
import { renderHook, act } from '@testing-library/react-hooks';
import AppLayout from '../components/App';
import { useVideoInfo, useVideList } from '../utils/hooks/useVideoStates';
import VideoMock from '../utils/mock/youtube-videos-mock.json';

afterEach(cleanup);

const MockVideoInfo = VideoMock.items[1].snippet;

it('Should render App', () => {
  act(() => {
    const app = render(<AppLayout />);
    expect(app).toMatchSnapshot();
  });
  const homePage = screen.getByTestId('yt-videocard');
  const headerMenu = screen.getByTestId('yt-header');
  expect(homePage).toBeInTheDocument();
  expect(headerMenu).toBeInTheDocument();
});

test('Test useVideoInfo Hook', () => {
  const { result } = renderHook(() => useVideoInfo());
  expect(result.current.video).toEqual({});
  expect(typeof result.current.updateVideoInfo).toBe('function');

  act(() => {
    result.current.updateVideoInfo(MockVideoInfo);
  });
  expect(result.current.video).toEqual({
    channelId: 'UCPGzT4wecuWM0BH9mPiulXg',
    channelTitle: 'Wizeline',
    description:
      'Follow Hector Padilla, Wizeline Director of Engineering, for a lively tour of our office. In 2018, Wizeline opened its stunning new office in Guadalajara, Jalisco, ...',
    liveBroadcastContent: 'none',
    publishTime: '2019-09-30T23:54:32Z',
    publishedAt: '2019-09-30T23:54:32Z',
    thumbnails: {
      default: {
        height: 90,
        url: 'https://i.ytimg.com/vi/nmXMgqjQzls/default.jpg',
        width: 120,
      },
      high: {
        height: 360,
        url: 'https://i.ytimg.com/vi/nmXMgqjQzls/hqdefault.jpg',
        width: 480,
      },
      medium: {
        height: 180,
        url: 'https://i.ytimg.com/vi/nmXMgqjQzls/mqdefault.jpg',
        width: 320,
      },
    },
    title: 'Video Tour | Welcome to Wizeline Guadalajara',
  });
});

test('Test useVideoInfo Hook', () => {
  const { result } = renderHook(() => useVideList());
  expect(result.current.videoList).toEqual([]);
  expect(typeof result.current.updateVideoList).toBe('function');
  expect(result.current.error).toEqual(null);
  expect(typeof result.current.setErrorMsg).toBe('function');
  act(() => {
    result.current.updateVideoList([
      VideoMock.items[0],
      VideoMock.items[1],
      VideoMock.items[3],
    ]);
  });
  const videoList = [VideoMock.items[0], VideoMock.items[1], VideoMock.items[3]];
  expect(result.current.videoList).toEqual(videoList);
});
