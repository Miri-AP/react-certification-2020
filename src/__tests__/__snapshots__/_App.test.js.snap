// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render App 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="sc-bdfBwQ hIHNKi"
        data-testid="yt-header"
      >
        <div
          class="sc-gsTCUz ywWHk"
        >
          <button
            class="sc-dlfnbm cOuLkH"
          />
          <div
            class="sc-hKgILt ljSDGR"
          >
            <input
              class="sc-eCssSg iBGkVj"
              id="name"
              placeholder="Search..."
              type="text"
              value="wizeline"
            />
            <span
              class="sc-jSgupP hcyBed"
            />
          </div>
        </div>
        <div
          class="sc-gsTCUz ywWHk"
        >
          <div
            class="sc-gKsewC kjfNIn"
          >
            <input
              class="sc-pFZIQ gGzfcK"
              id="darkMode"
              name="darkMode"
              type="checkbox"
            />
            <label
              class="sc-iBPRYJ cpTqSp"
              for="darkMode"
            >
              Dark mode
            </label>
          </div>
          <button
            class="sc-fubCfw dDcTFR"
          />
        </div>
      </header>
      <main
        class="container"
        style="background: rgb(255, 255, 255);"
      >
        <div
          class="sc-iwyYcG eTwARz"
          data-testid="user-login"
        >
          <form
            class="sc-cxFLnm jYsNLu"
          >
            <div
              class="form-group"
            >
              <label
                class="sc-iJuUWI fYFISK"
                for="username"
              >
                <p
                  class="sc-giIncl jxdTZO"
                >
                  Username
                </p>
                <input
                  class="sc-lmoMRL fKJZfF"
                  id="username"
                  placeholder="username"
                  required=""
                  type="text"
                  value=""
                />
              </label>
            </div>
            <div
              class="form-group"
            >
              <label
                class="sc-iJuUWI fYFISK"
                for="password"
              >
                <p
                  class="sc-giIncl jxdTZO"
                >
                  Password
                </p>
                <input
                  class="sc-lmoMRL fKJZfF"
                  id="password"
                  placeholder="password"
                  required=""
                  type="password"
                  value=""
                />
              </label>
            </div>
            <button
              class="sc-ezrdKe cgZrpU"
              data-testid="user-login-button"
              type="submit"
            >
              Log in
            </button>
          </form>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <header
      class="sc-bdfBwQ hIHNKi"
      data-testid="yt-header"
    >
      <div
        class="sc-gsTCUz ywWHk"
      >
        <button
          class="sc-dlfnbm cOuLkH"
        />
        <div
          class="sc-hKgILt ljSDGR"
        >
          <input
            class="sc-eCssSg iBGkVj"
            id="name"
            placeholder="Search..."
            type="text"
            value="wizeline"
          />
          <span
            class="sc-jSgupP hcyBed"
          />
        </div>
      </div>
      <div
        class="sc-gsTCUz ywWHk"
      >
        <div
          class="sc-gKsewC kjfNIn"
        >
          <input
            class="sc-pFZIQ gGzfcK"
            id="darkMode"
            name="darkMode"
            type="checkbox"
          />
          <label
            class="sc-iBPRYJ cpTqSp"
            for="darkMode"
          >
            Dark mode
          </label>
        </div>
        <button
          class="sc-fubCfw dDcTFR"
        />
      </div>
    </header>
    <main
      class="container"
      style="background: rgb(255, 255, 255);"
    >
      <div
        class="sc-iwyYcG eTwARz"
        data-testid="user-login"
      >
        <form
          class="sc-cxFLnm jYsNLu"
        >
          <div
            class="form-group"
          >
            <label
              class="sc-iJuUWI fYFISK"
              for="username"
            >
              <p
                class="sc-giIncl jxdTZO"
              >
                Username
              </p>
              <input
                class="sc-lmoMRL fKJZfF"
                id="username"
                placeholder="username"
                required=""
                type="text"
                value=""
              />
            </label>
          </div>
          <div
            class="form-group"
          >
            <label
              class="sc-iJuUWI fYFISK"
              for="password"
            >
              <p
                class="sc-giIncl jxdTZO"
              >
                Password
              </p>
              <input
                class="sc-lmoMRL fKJZfF"
                id="password"
                placeholder="password"
                required=""
                type="password"
                value=""
              />
            </label>
          </div>
          <button
            class="sc-ezrdKe cgZrpU"
            data-testid="user-login-button"
            type="submit"
          >
            Log in
          </button>
        </form>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Should render Favorites 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        data-testid="favorites-view"
      >
        <h1
          class="sc-kLgntA gYAoXa"
        >
          Your favorites
        </h1>
        <p>
          0
           favorite videos
        </p>
        <section
          class="sc-bYEvPH dCExnP"
        />
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      data-testid="favorites-view"
    >
      <h1
        class="sc-kLgntA gYAoXa"
      >
        Your favorites
      </h1>
      <p>
        0
         favorite videos
      </p>
      <section
        class="sc-bYEvPH dCExnP"
      />
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Should render NotFound 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="not-found"
        data-testid="not-found"
      >
        <a
          class="home-link"
          to="/"
        >
          home
        </a>
        <img
          alt="page not found"
          src="404.gif"
        />
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="not-found"
      data-testid="not-found"
    >
      <a
        class="home-link"
        to="/"
      >
        home
      </a>
      <img
        alt="page not found"
        src="404.gif"
      />
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
